pynxtools_xps/scienta/scienta_txt.py

```diff
@@ -201,15 +305,18 @@ def _update_xps_dict_with_spectrum(
 
         for grouping, spectrum_keys in template_key_map.items():
             root = path_map[str(grouping)]
+
             for spectrum_key in spectrum_keys:
+                mpes_key = spectrum_key.rsplit(" ", 1)[0]
                 try:
-                    units = re.search(r"\[([A-Za-z0-9_]+)\]", spectrum_key).group(1)
-                    mpes_key = spectrum_key.rsplit(" ", 1)[0]
-                    self._xps_dict[f"{root}/{mpes_key}/@units"] = units
-                    self._xps_dict[f"{root}/{mpes_key}"] = spectrum[spectrum_key]
-                except AttributeError:
-                    mpes_key = spectrum_key
                     self._xps_dict[f"{root}/{mpes_key}"] = spectrum[spectrum_key]
+                except KeyError as e:
+                    pass
+
+                unit_key = f"{grouping}/{spectrum_key}"
+                units = get_units_for_key(unit_key, UNITS)
+                if units:
```

Comment:
This is related to the comment on `get_units_for_key`. I would return `None` if there was no unit and check for `if units is None`. That way you can also write empty units for unit less values.

Reply:
Ok, but then the check should be `if units is not None`, right?